### PR TITLE
Login/Register pages done

### PR DESCRIPTION
**Pull request by hgvo9**
With this pull request, the functionnality of the login and register pages are done, I've done enough testing on my own but of course there are still bugs that can occurs that I might not have spot yet. Once I've finished developping the tests it will be much easier to test the pages to find bugs.

Major things to understand from this pull request is :

- Validators are implemented in login/register pages with validator.js (https://validatejs.org/)
- Alert messages shows up if you failed registering or logging
- Red messages appears at the bottom of the input texts if the validators are triggered (ex: wrong format of email, password too short, etc...)
- Redux-Persist implemented for persisted storage through sessions
- Private/Public routes are implemented in a way that when the users logs in they are automatically redirected to the homepage, thanks to conditionally rendering in react-navigation
- Added a reducers named commons where fetching states are stored

### CODE REVIEW

File: src/services/user.services.ts [hunk 1]
```diff
@@ -10,22 +10,22 @@ async function login({
...
+    .post('https://api.safely-app.fr/login', {
       email: username,
       password,
     })
     .then(res => {
       return res.data;
     })
     .catch(err => {
-      return err;
+      throw err;
```
Comment (louischristner) [inline src/services/user.services.ts:21]: Are you catching these exceptions somewhere ? I found it for the logout but not for the login and register functions.

**Review by louischristner — APPROVED** [review]
It looks good to me, great job !

File: src/services/user.services.ts (hunk 1 above)
Comment (hgvo9) [reply src/services/user.services.ts:22]: Yes, I am catching these exceptions when calling the api calls from the services in actions :

image

If the API call went wrong, I will catch it and dispatch an action failure which will be storing an error message in my state. Then in my component I'll be checking if anything went wrong after calling the api like that :

image

I am parsing the array 'error' which is the state where I'll be storing error messages if the fetching went wrong, if there is something wrong my component will do something and then I'll be resetting my state containing error messages for future calls

File: src/services/user.services.ts [hunk 2]
```diff
@@ -10,22 +10,22 @@ async function login({
   password: string;
 }): Promise<UserLoginInterface> {
   const response = await axios
-    .post('http://api.safely-app.fr/login', {
+    .post('https://api.safely-app.fr/login', {
```
[the remark this answers is not in the document]
Comment (hgvo9) [reply src/services/user.services.ts]: @CptHash I've added the .env file in my last commit, feel free to review it if needed

**Comment by hgvo9** [comment]
Just to clarify, the tests will be successfull once the pull request from the units-tests will be merged into the dev branch, currently in the dev branch there are no tests hence the failed test

**Pull request merged**